Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1832,19 +1832,15 @@ Optional<TimePercentage> Parser::parse_time_percentage(TokenStream<ComponentValu
     return {};
 }
 
-Optional<LengthOrCalculated> Parser::parse_source_size_value(ComponentValue const& component_value)
+Optional<LengthOrCalculated> Parser::parse_source_size_value(TokenStream<ComponentValue>& tokens)
 {
-    if (component_value.is_ident("auto"sv)) {
+    if (tokens.peek_token().is_ident("auto"sv)) {
+        (void)tokens.next_token(); // auto
         return LengthOrCalculated { Length::make_auto() };
     }
 
-    if (auto calculated_value = parse_calculated_value(component_value)) {
-        return LengthOrCalculated { calculated_value.release_nonnull() };
-    }
-
-    if (auto length = parse_length(component_value); length.has_value()) {
-        return LengthOrCalculated { length.release_value() };
-    }
+    if (auto length = parse_length(tokens); length.has_value())
+        return length;
 
     return {};
```

Comment:
Could this be consolidated to:
```suggestion
    return parse_length(tokens);
```

Reply:
You're right! Originally I had a transaction in there but removed it, and forgot to simplify this.